### PR TITLE
[FIX] account: do not open blindly any action

### PR DESCRIPTION
**Pull request by mart-e**
The `action_name` was taken from the context and needs sanity checks.
Ensure only actions from the account module can be read.

### CODE REVIEW

File: addons/account/models/account_journal_dashboard.py [hunk 1]
```diff
@@ -520,14 +521,14 @@ def open_action_with_context(self):
...
-        [action] = self.env[model].browse(action_id).read()
+        action = self.env['ir.action.act_window']._for_xml_id(f"account.{action_name}")
+        self.env[action['res_model']].check_access_rights('read')
+
         action['context'] = ctx
         if ctx.get('use_domain', False):
             action['domain'] = isinstance(ctx['use_domain'], list) and ctx['use_domain'] or ['|', ('journal_id', '=', self.id), ('journal_id', '=', False)]
             action['name'] = _(
-                "%(action)s for journal %(journal)s",
+                _("%(action)s for journal %(journal)s"),
```
Comment (KangOl) [inline addons/account/models/account_journal_dashboard.py:531]: It's already a `_()` call.

Comment (mart-e) [reply addons/account/models/account_journal_dashboard.py:531]: Needs... more... translations !

**Comment by mart-e** [comment]
@robodoo r+

File: odoo/tools/test_reports.py [hunk 2]
```diff
@@ -121,7 +121,7 @@ def log_test(msg, *args):
         act_model, act_id = action._name, action.id
     else:
         assert isinstance(action_id, int)
-        act_model = 'ir.action.act_window'     # assume that
+        act_model = 'ir.actions.act_window'     # assume that
         act_id = action_id
         act_xmlid = '<%s>' % act_id
```
Comment (xmo-odoo) [inline odoo/tools/test_reports.py:126]: It's only used in one place, maybe remove the `our_module` and `action_id: int` stuff entirely?

File: odoo/tools/test_reports.py [hunk 3]
```diff
@@ -74,7 +74,7 @@ def try_report(cr, uid, rname, ids, data=None, context=None, our_module=None, re
 def try_report_action(cr, uid, action_id, active_model=None, active_ids=None,
                 wiz_data=None, wiz_buttons=None,
                 context=None, our_module=None):
```
Comment (xmo-odoo) [inline odoo/tools/test_reports.py:76]: By `our_module` I meant this, if there's only one caller we could just update `action_id` to include the module name (if that's not already the case).

File: odoo/tools/test_reports.py (hunk 2 above)
Comment (mart-e) [reply odoo/tools/test_reports.py:126]: maybe a bit out of scope, this one was already found by grep but looks like dead code anyway

**Comment by mart-e** [comment]
@robodoo r+

**Pull request closed without merging**